contracts/modules/LoanSettings/LoanSettings.sol

```diff
@@ -93,6 +95,99 @@ contract LoanSettings is State, InterestHandler, LoanSettingsEvents {
         }
     }
 
+    function migrateLoanParamsList(
+        address owner,
+        uint256 start,
+        uint256 count)
+        external
+        onlyGuardian
+    {
+        EnumerableBytes32Set.Bytes32Set storage set = userLoanParamSets[owner];
+        uint256 end = start.add(count).min256(set.length());
+        if (start >= end) {
+            return;
+        }
+        count = end-start;
+
+        bytes32 loanParamId;
+        LoanParams memory loanParamsLocal;
+
+        for (uint256 i = start; i < end; ++i) {
+            loanParamsLocal = loanParams[set.get(i)];
+            loanParamId = getLoanParam(
+                loanParamsLocal.loanToken,
+                loanParamsLocal.collateralToken,
+                loanParamsLocal.maxLoanTerm == 0 // isTorqueLoan
+                    ? true
+                    : false
+            );
+            loanParamsIds[loanParamId] = loanParamsLocal.id;
+        }
+    }
+
+    function getLoanParam(
+        address loanToken,
+        address collateralToken,
+        bool isTorqueLoan)
+        pure
+        internal
+        returns(bytes32)
+    {
+        return keccak256(abi.encodePacked(
+                    loanToken,
+                    collateralToken,
+                    isTorqueLoan
+                ));
+    }
+
+    // This function intends to be PUBLIC so that anyone can create params if loanToken and collateralToken are approved by the protocol
+    function createDefaultParams(
```

Comment:
is this function necessary? getDefaultLoanParams() seems to be sufficient as if no ID is available it just auto plugs in the default settings.

Reply:
1) `getDefaultLoanParams` builds list on the fly. I was thinking if we see it used often we can actually save in storage
2) if we add more collateral tokens we would allow people to create default params by themselves

Reply:
hm. I am not sure if SLOAD is cheaper than just writing things in memory as all the things it writes into the blank loanParams are already loaded into memory anyways so I think loading from storage is the same cost. I think it would be best to only write it to storage if there is a unique param

Reply:
nevermind, this should be cheaper in long run.